core/src/test/scala/unit/kafka/cluster/PartitionTest.scala

```diff
@@ -2650,7 +2647,7 @@ class PartitionTest extends AbstractPartitionTest {
     assertEquals(Some(0L), partition.leaderEpochStartOffsetOpt)
 
     val leaderLog = partition.localLogOrException
-    assertEquals(Some(EpochEntry(leaderEpoch, 0L)), leaderLog.leaderEpochCache.flatMap(_.latestEntry))
+    assertEquals(Some(new EpochEntry(leaderEpoch, 0L)), leaderLog.leaderEpochCache.flatMap(_.latestEntry.asScala))
```

Comment:
We can change the expected value to be `Optional.of` instead of converting the response with `asScala`.